### PR TITLE
GoDB deployment + molecule

### PR DESCRIPTION
**Pull request by AaronJackson**

### CODE REVIEW

**Comment by AaronJackson** [comment]
Hmm "it works on my machine"

Possibly an issue with trying to run a container from inside the molecule container?  The whole thing might be wrapped inside a container on the GitHub runner too...

```
TASK [hic.godb : build container] **********************************************
  fatal: [instance]: FAILED! => {"changed": true, "cmd": "cd container\ndocker build -t godb .\n", "delta": "0:00:01.950703", "end": "2022-06-17 13:57:07.597556", "msg": "non-zero return code", "rc": 1, "start": "2022-06-17 13:57:05.646853", "stderr": "error creating aufs mount to /var/lib/docker/aufs/mnt/8b4155ca9096efdd87ce8e1efc3b878a94de7ffad24e9ea544dc1008c17d6e86-init: mount target=/var/lib/docker/aufs/mnt/8b4155ca9096efdd87ce8e1efc3b878a94de7ffad24e9ea544dc1008c17d6e86-init data=br:/var/lib/docker/aufs/diff/8b4155ca9096efdd87ce8e1efc3b878a94de7ffad24e9ea544dc1008c17d6e86-init=rw:/var/lib/docker/aufs/diff/8715d616583e89795fd98708e04ec2b991405bc0f35a0069f96719a16668de60=ro+wh,dio,xino=/dev/shm/aufs.xino: invalid argument", "stderr_lines": ["error creating aufs mount to /var/lib/docker/aufs/mnt/8b4155ca9096efdd87ce8e1efc3b878a94de7ffad24e9ea544dc1008c17d6e86-init: mount target=/var/lib/docker/aufs/mnt/8b4155ca9096efdd87ce8e1efc3b878a94de7ffad24e9ea544dc1008c17d6e86-init data=br:/var/lib/docker/aufs/dif...
  PLAY RECAP *********************************************************************
  instance                   : ok=10  changed=7    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
  WARNING  Retrying execution failure 2 of: ansible-playbook --inventory /home/runner/.cache/molecule/godb/default/inventory --skip-tags molecule-notest,notest /home/runner/work/shared-services-ansible-roles/shared-services-ansible-roles/godb/molecule/default/converge.yml
  CRITICAL Ansible return code was 2, command was: ['ansible-playbook', '--inventory', '/home/runner/.cache/molecule/godb/default/inventory', '--skip-tags', 'molecule-notest,notest', '/home/runner/work/shared-services-ansible-roles/shared-services-ansible-roles/godb/molecule/default/converge.yml']
  ::endgroup::
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,6 +54,9 @@ jobs:
       - name: Install ansible and molecule
         run: pip3 install ansible molecule[docker] docker
 
+      - name: Install ansible-galaxy dependencies
+        run: ansible-galaxy collection install community.mongodb:1.3.0 --force
```
Comment (manics) [inline .github/workflows/test.yml:58]: Doesn't molecule handle this (`dependencies.name: galaxy` with `requirements.yml`)?

File: godb/.yamllint [hunk 2]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (manics) [inline godb/.yamllint:1]: Do you think we should have one of these per role? I've moved this file to the root of the repo in https://github.com/hic-infra/shared-services-ansible-roles/pull/3

File: godb/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+
+galaxy_info:
+  author: Aaron Jackson
+  namespace: hic
+  description: |
+    Sets up GoDB running in a Docker container with httpd reverse proxy
+  company: Health Informatics Centre, University of Dundee
+
+  license: license (GPL-2.0-or-later, MIT, etc)
```
Comment (manics) [inline godb/meta/main.yml:10]:
```suggestion
  license: MIT
```
Otherwise this will fail ansible-lint

File: godb/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+  min_ansible_version: 2.1
+
+  platforms:
+    - name: Ubuntu
+      versions:
+        - 20.04
```
Comment (manics) [inline godb/meta/main.yml:17]:
```suggestion
        - focal
```

File: godb/molecule/default/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
+dependency:
+  name: galaxy
+driver:
+  name: docker
+platforms:
+  - name: instance
+    image: mpaivabarbosa/molecule-systemd-ubuntu
```
Comment (manics) [inline godb/molecule/default/molecule.yml:8]:
```suggestion
    image: docker.io/mpaivabarbosa/molecule-systemd-ubuntu
```

File: godb/tasks/docker.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+---
```
Comment (manics) [inline godb/tasks/docker.yml:1]: This could be a candidate for a separate role in future? But fine for now.

File: godb/tasks/godb.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      WantedBy=multi-user.target
+
+- meta: flush_handlers
+
+- name: clone godb git repo
+  register: clone
+  git:
+    repo: https://github.com/hicresearch/GoDb
+    dest: /home/ubuntu/GoDB
+    version: deploy
```
Comment (manics) [inline godb/tasks/godb.yml:28]: Do you reckon we can start tagging GoDb? Or are there still too many pending changes?

File: .github/workflows/test.yml (hunk 1 above)
Comment (AaronJackson) [reply .github/workflows/test.yml:58]: I really hoped so but whenever I removed it, it failed due to a recent change in the mongodb collection. Unless you can see I'm doing something wrong else-where with Ansible requirements management.

File: godb/.yamllint (hunk 2 above)
Comment (AaronJackson) [reply godb/.yamllint:1]: I think a shared one is good! I can remove this.

File: godb/tasks/docker.yml (hunk 6 above)
Comment (AaronJackson) [reply godb/tasks/docker.yml:1]: agreed!

File: .github/workflows/test.yml (hunk 1 above)
Comment (manics) [reply .github/workflows/test.yml:67]: Maybe there's a bug somewhere 😢

**Pull request merged**